braindecode/augmentation/base.py

```diff
@@ -0,0 +1,208 @@
+# Authors: Cédric Rommel <cpe.rommel@gmail.com>
+#
+# License: BSD (3-clause)
+
+from typing import List, Tuple, Any
+from numbers import Real
+from collections.abc import Iterable
+
+import pandas as pd
+from sklearn.utils import check_random_state
+import torch
+from torch import Tensor, nn
+from torch.utils.data import DataLoader
+from torch.utils.data._utils.collate import default_collate
+
+from braindecode.augmentation.functionals import identity
+
+Batch = List[Tuple[torch.Tensor, int, Any]]
+Output = Tuple[torch.Tensor, torch.Tensor]
+
+
+class Transform(torch.nn.Module):
+    """ Basic transform class used for implementing data augmentation
+    operations
+
+    Parameters
+    ----------
+    operation : callable
+        A function taking arrays X, y (sample features and
+        target resp.) and other required arguments, and returning the
+        transformed X and y.
+    probability : float, optional
+        Float between 0 and 1 defining the uniform probability of
+        applying the operation. Set to 1.0 by default (e.g always apply the
+        operation).
+    magnitude : float | None, optional
+        Defines the strength of the transformation applied between 0 and 1 and
+        depends on the nature of the transformation and on its range. Some
+        transformations don't have any magnitude (=None). It can be equivalent
+        to another argument of object with more meaning. In case both are
+        passed, magnitude will override the latter. Defaults to None.
+    mag_range : tuple of two floats | None, optional
+        Valid range of the argument mapped by `magnitude` (e.g. standard
+        deviation, number of sample, etc.):
+        ```
+        argument = magnitude * mag_range[1] + (1 - magnitude) * mag_range[0].
+        ```
+        If `magnitude` is None it is ignored. Defaults to None.
+    random_state: int, optional
+        Seed to be used to instatiate numpy random number generator instance.
+        Used to decide whether or not to transform given the probability
+        argument. Defaults to None.
```

Comment:
possible interference with `set_random_seeds`?

Reply:
As discussed, it does not interfere with `set_random_seeds` and just complements it, so I guess I'll leave it here unless it raises some issues :)